### PR TITLE
fixes #2878 - Import Classes fails when using parser = future

### PR DESCRIPTION
**Pull request by rickard-von-essen**
This enables _smart-proxy_ to read puppet classes which contains new _lambda_ constructs when `parser = future` option is used.

For more information see:
http://docs.puppetlabs.com/puppet/3/reference/experiments_future.html
http://projects.theforeman.org/issues/2878

### CODE REVIEW

**Comment by ohadlevy** [comment]
[Test]

File: lib/proxy/puppet/environment.rb [hunk 1]
```diff
@@ -122,8 +122,11 @@ def to_s
...
-      paths.map {|path| PuppetClass.scan_directory path}.flatten
+      conf = Puppet.settings.instance_variable_get(:@values)
+      eparser = conf[:master] && conf[:master][:parser] == 'future'
+
+      paths.map {|path| PuppetClass.scan_directory path, eparser}.flatten
     end
 
   end
-end
+end
```
Comment (ekohl) [inline lib/proxy/puppet/environment.rb]: Nitpick, but I'd keep the line ending here.

Comment (rickard-von-essen) [inline lib/proxy/puppet/environment.rb]: Fixed!

**Comment by ohadlevy** [comment]
[test]

**Comment by domcleal** [comment]
This is awesome, thanks for the contribution @rickard-von-essen.  I fixed a couple of weird issues under Ruby 1.8.7 and Puppet 3.2+, but all merged into af58b9f for Foreman 1.4.

**Pull request closed without merging**